base/ordering.jl

```diff
@@ -125,18 +125,13 @@ lt(o::Lt,                    a, b) = o.lt(a,b)
     (lt(p.order, da, db)::Bool) | (!(lt(p.order, db, da)::Bool) & (a < b))
 end
 
-_ord(lt::typeof(isless), by::typeof(identity), order::Ordering) = order
-_ord(lt::typeof(isless), by,                   order::Ordering) = By(by, order)
-
-function _ord(lt, by, order::Ordering)
-    if order === Forward
-        return Lt((x, y) -> lt(by(x), by(y)))
-    elseif order === Reverse
-        return Lt((x, y) -> lt(by(y), by(x)))
-    else
-        error("Passing both lt= and order= arguments is ambiguous; please pass order=Forward or order=Reverse (or leave default)")
-    end
-end
+
+_ord(lt::typeof(isless), by, order::Ordering)        = _by(by, order)
+_ord(lt,                 by, order::ForwardOrdering) = _by(by, Lt(lt))
+_ord(lt,                 by, order::ReverseOrdering{ForwardOrdering}) = reverse(_by(by, Lt(lt)))
+_ord(lt,                 by, order::Ordering) = error("Passing both lt= and order= arguments is ambiguous; please pass order=Forward or order=Reverse (or leave default)")
+_by(by, order::Ordering) = By(by, order)
+_by(::typeof(identity), order) = order
```

Comment:
This is ambiguous for `_ord(isless, ::Any, Forward)` and `_ord(isless, ::Any, Reverse)` (as was my suggestion above). Perhaps add two specific methods for those two cases?

<!--, or maybe better, define 
`_ord(lt, by, order) = _by(by, _lt(lt, order))`
and let `_lt` handle the case where lt and order are both provided:
```suggestion
_ord(lt, by, order) = _by(by, _lt(lt, order))
_by(by, order::Ordering) = By(by, order)
_by(::typeof(identity), order::Ordering) = order
_lt(::typeof(isless), order::Ordering) = order
_lt(lt, ::ForwardOrdering) = Lt(lt)
_lt(lt, ::ReverseOrdering{ForwardOrdering}) = reverse(Lt(lt))
_lt(::typeof(isless), ::ForwardOrdering) = Forward
_lt(::typeof(isless), ::ReverseOrdering{ForwardOrdering}) = Reverse
_lt(lt, order::Ordering) = error("Passing both lt= and order= arguments is ambiguous; please pass order=Forward or order=Reverse (or leave default)")-->